File: 백준/Silver/15666. N과 M （12）/N과 M （12）.py
import sys
input = sys.stdin.readline

N, M = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()

s_arr = []
for i in arr:
    if i in s_arr:
        continue
    else:
        s_arr.append(i)

def back(cnt, sm):

    if (cnt == M):
        print(*sm)
        return

    for i in s_arr:
        if ((sm and sm[-1] <= i) or len(sm) == 0):
            sm.append(i)
            back(cnt+1, sm)
            sm.pop()

back(0, [])

File: 백준/Silver/15666. N과 M （12）/N과 M （12）.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");

const [N, M] = input[0].split(" ").map((x) => Number(x));

const arr = input[1].split(" ").map((x) => Number(x));
arr.sort((a, b) => a - b);

const s_arr = [];

for (let i = 0; i < N; i++) {
    if (s_arr.includes(arr[i])) continue;
    else s_arr.push(arr[i]);
}

const result = [];

function back(cnt, sm) {
    if (cnt === M) {
        result.push(sm.join(" "));
        return;
    }

    for (let i = 0; i < N; i++) {
        if ((sm.length && sm[sm.length - 1] <= s_arr[i]) || sm.length == 0) {
            sm.push(s_arr[i]);
            back(cnt + 1, sm);
            sm.pop();
        }
    }
}

back(0, []);
console.log(result.join("\n"));
